engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/CurryingTest.scala

```diff
@@ -74,11 +73,63 @@ class CurryingTest extends InterpreterTest {
           |    fn1 = Nothing.fn ...
           |    fn2 = fn1 1 2 ...
           |    fn3 = fn2 3 ...
-          |
-          |    fn3.call
+          |    fn3
           |""".stripMargin
 
       eval(code) shouldEqual 26
     }
+
+    "automatically force functions with all-defaulted arguments" in {
+      val code =
+        """main =
+          |     foo (x=1) = x
+          |     foo + 1
+          |""".stripMargin
+      eval(code) shouldEqual 2
+    }
+
+    "allow to pass suspended functions in arguments with `...`" in {
+      val code =
+        """main =
+          |    foo f = f 1
+          |    bar x=1 = x + 1
+          |    foo (bar ...)
```

Comment:
Minor nitpick but I'd suggest making the argument passed to `f` have different value than the default one - this way we ensure that we do not get some weird behaviour where the default one was used instead (it seems _extremely_ unlikely to get a bug with such behaviour, but you never know and the cost of testing this case is 0 as we just need to change this value).

Reply:
Good point, I wouldn't call it "extremely unlikely" given the behavior I saw in the process of implementing this, and I would call the test as it exists "extremely bad at testing things" :P